nestkernel/target.h

```diff
@@ -75,163 +114,197 @@ class Target
-  data_ = ( data_ & ( ~lcid_mask ) )
-    | ( static_cast< unsigned long >( lcid ) << lcid_shift );
+  assert( lcid <= MAX_LCID );
+  remote_target_id_ = ( remote_target_id_ & ( ~MASK_LCID ) )
+    | ( static_cast< uint64_t >( lcid ) << BITPOS_LCID );
 }
 
 inline index
 Target::get_lcid() const
 {
-  return ( data_ & lcid_mask ) >> lcid_shift;
+  return ( ( remote_target_id_ & MASK_LCID ) >> BITPOS_LCID );
 }
 
 inline void
 Target::set_rank( const thread rank )
 {
-  assert( rank < max_rank_ );
-  data_ = ( data_ & ( ~rank_mask ) )
-    | ( static_cast< unsigned long >( rank ) << rank_shift );
+  assert( rank <= MAX_RANK );
+  remote_target_id_ = ( remote_target_id_ & ( ~MASK_RANK ) )
+    | ( static_cast< uint64_t >( rank ) << BITPOS_RANK );
 }
 
 inline thread
 Target::get_rank() const
 {
-  return ( data_ & rank_mask ) >> rank_shift;
+  return ( ( remote_target_id_ & MASK_RANK ) >> BITPOS_RANK );
 }
 
 inline void
 Target::set_tid( const thread tid )
 {
-  assert( tid < max_tid_ );
-  data_ = ( data_ & ( ~tid_mask ) )
-    | ( static_cast< unsigned long >( tid ) << tid_shift );
+  assert( tid <= MAX_TID );
+  remote_target_id_ = ( remote_target_id_ & ( ~MASK_TID ) )
+    | ( static_cast< uint64_t >( tid ) << BITPOS_TID );
 }
 
 inline thread
 Target::get_tid() const
 {
-  return ( data_ & tid_mask ) >> tid_shift;
+  return ( ( remote_target_id_ & MASK_TID ) >> BITPOS_TID );
 }
 
 inline void
 Target::set_syn_id( const synindex syn_id )
 {
-  assert( syn_id < max_syn_id_ );
-  data_ = ( data_ & ( ~syn_id_mask ) )
-    | ( static_cast< unsigned long >( syn_id ) << syn_id_shift );
+  assert( syn_id <= MAX_SYN_ID );
+  remote_target_id_ = ( remote_target_id_ & ( ~MASK_SYN_ID ) )
+    | ( static_cast< uint64_t >( syn_id ) << BITPOS_SYN_ID );
 }
 
 inline synindex
 Target::get_syn_id() const
 {
-  return ( data_ & syn_id_mask ) >> syn_id_shift;
+  return ( ( remote_target_id_ & MASK_SYN_ID ) >> BITPOS_SYN_ID );
 }
 
 inline void
-Target::set_is_processed( const bool processed )
+Target::set_status( enum_status_target_id set_status_to )
+{
+  switch ( set_status_to )
+  {
+  case TARGET_ID_PROCESSED:
+    remote_target_id_ =
+      remote_target_id_ | MASK_PROCESSED_FLAG; // set single bit
+    break;
+  case TARGET_ID_UNPROCESSED:
+    remote_target_id_ =
+      remote_target_id_ & ~MASK_PROCESSED_FLAG; // clear single bit
+    break;
+  default:
+    throw InternalError( "Invalid remote target id status." );
+  }
+}
+
+inline enum_status_target_id
+Target::get_status() const
 {
-  data_ = ( data_ & ( ~processed_mask ) )
-    | ( static_cast< unsigned long >( processed ) << processed_shift );
+  if ( ( remote_target_id_ & MASK_PROCESSED_FLAG )
+    >> BITPOS_PROCESSED_FLAG ) // test single bit
+  {
+    return ( TARGET_ID_PROCESSED );
+  }
+  return ( TARGET_ID_UNPROCESSED );
 }
 
 inline bool
 Target::is_processed() const
 {
-  return ( data_ & processed_mask ) >> processed_shift;
+  if ( Target::get_status() == TARGET_ID_PROCESSED )
```

Comment:
Guess it's a question of taste, I prefer the conciseness of `return Target::get_status() == TARGET_ID_PROCESSED`

Reply:
Yes, much better!